monitoring/src/test/scala/akkaviz/server/WebSocketReplTest.scala

```diff
@@ -0,0 +1,140 @@
+
+import akka.actor.ActorSystem
+import akka.http.scaladsl.model.ws.{BinaryMessage, Message, TextMessage}
+import akka.stream.ActorMaterializer
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.stream.testkit.TestSubscriber.OnNext
+import akka.stream.testkit.scaladsl.{TestSink, TestSource}
+import akka.util.ByteString
+import ammonite.ops.Path
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.{FunSuite, Matchers}
+
+import scala.concurrent.duration._
+
+class WebSocketReplTest extends FunSuite with ScalaFutures with Matchers {
+
+  private[this] implicit val system = ActorSystem()
+  private[this] implicit val materializer = ActorMaterializer()
+
+  override implicit val patienceConfig: PatienceConfig = PatienceConfig(10.seconds)
+
+  private[this] def createRepl(fn: Function2[InputStream, OutputStream, Unit]) = new WebSocketRepl {
+    override protected def startReplProcessing(
+      in: InputStream, out: OutputStream, err: OutputStream, homePath: Path
+    ): Unit = {
+      fn(in, out)
+    }
+  }
+
+  private[this] def wrongMsg = fail("Response should be BinaryMessage.Strict")
+
+  private[this] val pingPongAndClose: Function2[InputStream, OutputStream, Unit] = {
+    (in, out) =>
+      val ab = new Array[Byte](4)
+      in.read(ab)
+      ByteString(ab).utf8String shouldBe "ping"
+      out.write("pong".getBytes)
+      in.close()
+      out.flush()
+      out.close()
+  }
+
+  test("It can work in both ways") {
+    val wsRepl = createRepl(pingPongAndClose)
+    val res = Source.single(BinaryMessage.Strict(ByteString("ping"))).via(wsRepl.replWebsocketFlow).runWith(Sink.head)
+    whenReady(res) {
+      case BinaryMessage.Strict(data) =>
+        data.utf8String shouldBe "pong"
+      case _ => wrongMsg
+    }
+  }
+
+  test("It forwards exceptions from the repl") {
+    val wsRepl = createRepl {
+      (in, out) =>
+        throw new Exception("Failure in REPL")
+    }
+    val res = Source.single(BinaryMessage.Strict(ByteString("ping"))).via(wsRepl.replWebsocketFlow).runWith(Sink.head)
+    whenReady(res) {
+      case BinaryMessage.Strict(data) =>
+        data.utf8String should include("Failure in REPL")
+      case _ => wrongMsg
+    }
+  }
+
+  test("Repl accepts TextMessage too") {
+    val wsRepl = createRepl(pingPongAndClose)
+    val res = Source.single(TextMessage.Strict("ping")).via(wsRepl.replWebsocketFlow).runWith(Sink.head)
+    whenReady(res) {
+      case BinaryMessage.Strict(data) =>
+        data.utf8String shouldBe "pong"
+      case _ => wrongMsg
+    }
+  }
+
+  test("It sends Keep Alive") {
+    val wsRepl = new WebSocketRepl {
+      override protected def startReplProcessing(
+        in: InputStream, out: OutputStream, err: OutputStream, homePath: Path
+      ): Unit = {
+        pingPongAndClose(in, out)
+      }
+
+      override protected def replKeepAliveEvery: FiniteDuration = 10.millis
+
+      def getReplKeepAliveMessage: Message = super.replKeepAliveMessage
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Do not prefix getters with 'get'](https://www.codacy.com/app/lustefaniak/akka-viz/file/2329236809/issues/source?bid=3121282&fileBranchId=3206949#l89)